File: python/number_guess.py
import random
class NumberGuess(object):
  def __init__(self):
    self.thinking_number = random.randint(1, 100)
    print("I am thinking of a number between 1 and 100")
  
  def check(self, guess):
    if self.thinking_number == guess:
      return 0
    elif( self.thinking_number > guess):
      return -1
    else:
      return 1
      

print("Welcome to the Number Guessing Game")

ng = NumberGuess()

while(True):
  guess = input("Guess my Number: ")
  result = ng.check(guess)
  if result == 0:
    print("You Guessed It !!!")
    break
  elif result == -1:
    print("Higher...")
  else:
    print("Lower...")
    
    

File: ruby/number_guess.rb

class NumberGuess
  def initialize()
    r = Random.new()
    @thinking_number = r.rand(1..100)
    puts "I am thining of a number between 1 and 100"
  end
  def check guess
    if @thinking_number == guess
      result = 0
    elsif @thinking_number > guess
      result = -1
    else
      result = 1
    end
    result 
  end
end

puts "Welcome to the Number Guessing Game"

ng = NumberGuess.new
while true
  print "Guess My Number: "
  guess = gets.chomp
  result = ng.check guess.to_i
  if result == 0 
    puts "You Got it!!!"
    break
  elsif result == -1
    puts "Higher..."
  else
    puts "Lower..."
  end
end
